linera-chain/src/data_types.rs

```diff
@@ -450,8 +454,11 @@ pub struct Vote<T> {
 
 impl<T> Vote<T> {
     /// Use signing key to create a signed object.
-    pub fn new(value: Hashed<T>, round: Round, key_pair: &KeyPair) -> Self {
-        let hash_and_round = ValueHashAndRound(value.hash(), round);
+    pub fn new(value: Hashed<T>, round: Round, key_pair: &KeyPair) -> Self
+    where
+        T: CertificateValueT,
+    {
+        let hash_and_round = ValueHashAndRound(value.hash(), round, T::KIND);
```

Comment:
It's now hash and round _and kind_. Maybe just `VoteContent` or `VoteValue`?

```suggestion
        let content = VoteContent(value.hash(), round, T::KIND);
```